driver-sync/src/test/functional/com/mongodb/client/unified/Entities.java

```diff
@@ -529,30 +522,42 @@ private void initClient(final BsonDocument entity, final String id,
+                                env = "test";
+                            }
+                            MongoCredential oidcCredential = MongoCredential
+                                    .createOidcCredential(null)
+                                    .withMechanismProperty("ENVIRONMENT", env);
+                            if (env.equals("azure")) {
+                                oidcCredential = oidcCredential
+                                        .withMechanismProperty("TOKEN_RESOURCE", getenv("AZUREOIDC_RESOURCE"));
+                            } else if (env.equals("gcp")) {
+                                oidcCredential = oidcCredential
+                                        .withMechanismProperty("TOKEN_RESOURCE", getenv("GCPOIDC_RESOURCE"));
+                            }
+                            clientSettingsBuilder.credential(oidcCredential);
                             break;
                         }
                         throw new UnsupportedOperationException("Unsupported authMechanism: " + value);
                     case "authMechanismProperties":
-                        MongoCredential credential = clientSettingsBuilder.build().getCredential();
-                        boolean isOidc = credential != null
-                                && credential.getAuthenticationMechanism() == AuthenticationMechanism.MONGODB_OIDC;
-                        boolean hasPlaceholder = value.equals(new BsonDocument("$$placeholder", new BsonInt32(1)));
-                        if (isOidc && hasPlaceholder) {
-                            clientSettingsBuilder.credential(credential.withMechanismProperty(
-                                    MongoCredential.OIDC_CALLBACK_KEY,
-                                    (MongoCredential.OidcCallback) context -> {
-                                        Path path = Paths.get(getenv(OidcAuthenticator.AWS_WEB_IDENTITY_TOKEN_FILE));
-                                        String accessToken;
-                                        try {
-                                            accessToken = new String(Files.readAllBytes(path), StandardCharsets.UTF_8);
-                                        } catch (IOException e) {
-                                            throw new RuntimeException(e);
-                                        }
-                                        return new MongoCredential.OidcCallbackResult(accessToken, Duration.ZERO);
-                                    }));
-                            break;
+                        // authMechanismProperties are handled as part of authMechanism, above
+                        BsonValue authMechanism = entity
+                                .getDocument("uriOptions")
+                                .get("authMechanism");
+                        if (authMechanism.equals(new BsonString(MONGODB_OIDC.getMechanismName()))) {
+                            break; // only OIDC supports authMechanismProperties
```

Comment:
> only OIDC supports authMechanismProperties

OIDC is not the only mechanism that supports the `authMechanismProperties` connection string option. Let's remove the comment?

This connection string option have existed before OIDC, and can be specified in `uriOptions` for other authentication mechanisms in unified tests. However, in practice, all currently existing unified tests specify `authMechanismProperties` in `uriOptions` only for the OIDC mechanism, which is why `UnsupportedOperationException` is thrown below when the mechanism is not OIDC. We do this all the time, and will implement applying `authMechanismProperties` for other auth mechanisms in unified tests on demand.

Reply:
"authMechanismProperties only supported here for OIDC", or do you feel no comment is needed at all? I am trying to note why this is `break`ing for OIDC but not others.

Reply:
I would not have the comment at all, as the fact that `break;` is executed only for OIDC is clear from the `if` condition, which checks exactly for that. But the updated comment does not have a second meaning (is not ambiguous), as the previous comment, so if you think it's needed, I have no objections.